script/make_release.py

```diff
@@ -0,0 +1,392 @@
+                    help='Sometimes too many compilation processes running in parallel can cause issues. If this is specified it will limit the number of parallel arch builds to this number. Remaining threads specified by concurrency will be used for make jobs.')
+parser.add_argument('--compiler', default=None, type=str,
+                    help='If specified this will be passed as COMP to the make script.')
+parser.add_argument('archs', nargs='*', default=None, type=str,
+                    help='A list of architectures to compile. If not provided then most sensible archs will be built.')
+
+args = parser.parse_args()
+
+if args.verbose and args.keep_logs:
+    parser.error('--verbose and --keep-logs are currently not compatible.')
+
+SDE_PATH = args.sde
+NUM_THREADS = args.concurrency
+CLONE_DIR = args.clone_dir
+OUT_DIR = args.out_dir
+REPO = args.repository
+BUILD_TYPE = args.build_type
+ARCHS = args.archs
+STDOUT = None if args.verbose else subprocess.DEVNULL
+STDERR = subprocess.STDOUT
+PREFIX = args.prefix
+IS_WINDOWS = 'windows' in platform.system().lower()
+EXE_SUFFIX = '.exe' if IS_WINDOWS else ''
+ZIP_SUFFIX = '.zip'
+EXPECTED_BENCH = args.bench
+MAX_NUM_WORKERS = NUM_THREADS if args.max_num_workers is None else args.max_num_workers
+COMPILER = args.compiler
+LOGGER = get_logger()
+
+def remove_readonly(func, path, _):
+    '''
+    Clear the readonly bit and reattempt the removal
+    '''
+    os.chmod(path, stat.S_IWRITE)
+    func(path)
+
+def clone_stockfish(destination_dir, repository, branch='master'):
+    '''
+    Clone, with depth 1, the specified `branch` from the `repository`.
+    The repository is cloned to the `destination_dir`.
+    '''
+    LOGGER.info('Cloning branch {} from repository {} into {}' \
+                .format(branch, repository, destination_dir))
+
+    subprocess.run( \
+        ['git', 'clone', '--branch', branch, '--depth', '1', repository, destination_dir], \
+        encoding='utf-8', stdout=STDOUT, stderr=STDERR)
+
+    LOGGER.info('Cloning finished.')
+
+class DummyContextManager():
+    '''
+    Useful for "conditional" `with` statements.
+    '''
+
+    def __enter__(self):
+        return None
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        return False
+
+class TemporaryDirectory():
+    '''
+    Maintains a temporary directory with a unique name.
+    '''
+
+    def __init__(self, prefix='', add_uuid=True):
+        self.path = prefix
+        if add_uuid:
+            self.path += str(uuid.uuid4())
+        os.mkdir(self.path)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        shutil.rmtree(self.path, onerror=remove_readonly)
+
+def get_bench_from_git_history(repo_dir):
+    command = ['git', '--git-dir', os.path.join(repo_dir, '.git'), 'log']
+    completed_process = subprocess.run(command, encoding='utf-8', capture_output=True, universal_newlines=True)
```

Comment:
`capture_output` is python 3.7, if I'm not wrong only one of `universal_newlines=True` or `encoding='utf-8'` is required to have a string output.

`completed_process = subprocess.run(command, stdout=subprocess.PIPE, universal_newlines=True)`

Reply:
right, I'll change this. But specifying `encoding` seems to still be required.